fs/osfs.py

```diff
@@ -261,17 +297,17 @@ def getinfo(self, path, namespaces=None):
 
     def listdir(self, path):
         self.check()
-        _path = self.validatepath(path)
-        sys_path = self._to_sys_path(_path)
+        sys_path = self._get_validated_syspath(path)
         with convert_os_errors('listdir', path, directory=True):
-            names = os.listdir(sys_path)
+            names = [fsdecode(f) for f in os.listdir(sys_path)]
         return names
 
     def makedir(self, path, permissions=None, recreate=False):
         self.check()
         mode = Permissions.get_mode(permissions)
+        path = path and fsdecode(path) or path
```

Comment:
There's a convention we use that you don't modify a `path` variable once its passed in. The reason is that when you raise an error containing the path, the error message will contain the same path that the developer called the method with.

Could you replace this pattern with `path = fsdecode(path) if path else path` ? It's more familiar for developers who came to Python recently.

Reply:
I can use a _path instead alright. But the and/or is more pythonic than if/else in my book. I can admit though that it may not be obvious ;)

Reply:
Fixed